Support setting a custom prefix to be used at runtime

timsueberkrueb opened the pull request:
To allow the Liri Text snap application to read the bundled
language specs at runtime, we need to be able to set
different paths for compile time and runtime.
This would be a possible implementation, let me know
if you'd like this to be done differently:
Support `CUSTOM_RUNTIME_DATADIR` and use it for
`ABSOLUTE_LANGUAGE_PATH`.
Rename `DATA_PREFIX` to `DATA_INSTALL_PREFIX` to clarify.

Aldrog commented on syntax-config.pri:
```diff
@@ -7,11 +7,16 @@ win32|macx {
+    
+    DATA_INSTALL_PREFIX = $$LIRI_INSTALL_DATADIR/liri-text
+    DATA_RUNTIME_PREFIX = $$CUSTOM_RUNTIME_DATADIR/liri-text
+    isEmpty(DATA_RUNTIME_PREFIX) {
```
You're checking DATA_RUNTIME_PREFIX for emptiness right after setting it. Following branch will never be active.

Aldrog replied:
I think it will be alright if you delete line 12.

timsueberkrueb replied:
@Aldrog oh, right, better check if `CUSTOM_RUNTIME_DATADIR` is defined.

Aldrog replied:
Why not to simply pass DATA_RUNTIME_PREFIX in special cases like snap packaging?

timsueberkrueb replied:
Okay, sure.

tim-sueberkrueb commented:
@Aldrog you might want to squash this.

tim-sueberkrueb commented:
Or, let me do it, don't merge yet.

timsueberkrueb committed:
Support setting a custom prefix to be used at runtime
Support DATA_RUNTIME_PREFIX and use it for
ABSOLUTE_LANGUAGE_PATH.
Rename DATA_PREFIX to DATA_INSTALL_PREFIX to clarify.

tim-sueberkrueb commented:
Thanks @Aldrog!

Aldrog commented:
Thank you for doing all the snap stuff :)
